### ISSUE TITLE
[Bug] `arc` generation uses unknown type of `ID` in generated classes

### ISSUE THREAD

**Issue by Smurf-IV** [pocoproject/poco#4126]
But it should be using `keyType(_class)` call.

i.e.
`<< "inline " << paramType(property) << " " << _class.name << "::" << property.name << "ID() const\n"`

Should be 
`<< "inline " << paramType(property) << " " << _class.name << "::" << property.name << keyType(_class) << "() const\n"`

**Comment by Smurf-IV** [comment]
Using the following
``` xml
<?xml version="1.0" encoding="utf-8"?>
<project convertCamelCase="false">
    <class name="CtrFiles" table="CtrFiles" key="TsId">
        <property name="TsId" type="uint64" />
        <property name="DateTime" type="dateTime"/>
        <property name="FileName" type="string"/>
    </class>
</project>
```

I would expect the following code to be generated
``` c++
CtrFiles::CtrFiles(const Poco::UInt64& id):
	Poco::ActiveRecord::ActiveRecord<Poco::UInt64>(id)
{
}
```

**Comment by obiltschnig** [comment]
You should set your include search paths properly.

**Comment by Smurf-IV** [comment]
> You should set your include search paths properly.

What? Arc generates the directories (Based on the existence of the namespace or not in the xml file)
The headers files include locations in the **generated source** files are wrong.!

**Comment by obiltschnig** [comment]
Because by our convention `include` and relative paths are never put into an `#include` statement.

**Comment by Smurf-IV** [comment]
This:
https://github.com/pocoproject/poco/blob/419433cf39dce3655540333902ab6dcab4821347/ActiveRecord/Compiler/src/Compiler.cpp#L166
Seems to contradict that statement.

**Comment by obiltschnig** [comment]
why?

**Comment by Smurf-IV** [comment]
That line places them into a directory (In the project) under `include`
But that is then not referenced from the cpp file to go find them in that place (Because they are under a `src` subdirectory ..

**Comment by obiltschnig** [comment]
Yes, because that's the exact same convention used for all of the Poco sources. And in non of the Poco .cpp files do we #include with "../include/Poco/Whatever.h". Because the build system sets an include search path.

**Comment by Smurf-IV** [comment]
I'm not disagreeing with the "Poco Build System".
These generated files go into your users projects, which could be using "hdr / filters / nothing / namespaces etc" as directories.
These generated files need to work together. That's all I'm saying.
Or for the code (User guide) to state that "./include" needs to be put into the using project search path.
